### DISCUSSION TITLE
Debouncing resolver

### DISCUSSION THREAD

**Discussion by asaf** [react-hook-form/react-hook-form#6027] [Q&A]
Hey,

We have a server-side validation for the whole form, so we use `Resolver` to check for errors asynchronously.
The problem is that resolver is invoked per field change, causing the server to perform validation per keystroke,

Is there a way to debounce the resolver to avoid a backend call per change?

Best,
Asaf.

**Comment by bluebill1049** [comment]
use `trigger` instead, manually fire validation.

**Reply by asaf** [reply]
so change `reValidateMode` to _onSubmit_, and `trigger` will invoke the `resolver`? if so, what is the correct way to invoke a trigger on every form change?

**Reply by bluebill1049** [reply]
`trigger` at `onChnage` and debounce there.
